src/sidebar/build-thread.js

```diff
@@ -203,96 +198,93 @@ function sort(threads, compareFn) {
+ *       comparison function for sorting replies
+ * @return {Thread}
  */
 function sortThread(thread, compareFn, replyCompareFn) {
-  const children = thread.children.map(function (child) {
-    return sortThread(child, replyCompareFn, replyCompareFn);
-  });
+  const children = thread.children.map(child =>
+    sortThread(child, replyCompareFn, replyCompareFn)
+  );
 
-  return Object.assign({}, thread, {
-    children: sort(children, compareFn),
-  });
+  return { ...thread, ...{ children: sort(children, compareFn) } };
 }
 
 /**
- * Return a copy of @p thread with the `replyCount` and `depth` properties
+ * Return a copy of thread with the `replyCount` and `depth` properties
  * updated.
+ *
+ * @param {Thread} thread
+ * @param {number} depth
+ * @return {Thread}
  */
 function countRepliesAndDepth(thread, depth) {
-  const children = thread.children.map(function (c) {
-    return countRepliesAndDepth(c, depth + 1);
-  });
-  return Object.assign({}, thread, {
-    children: children,
-    depth: depth,
-    replyCount: children.reduce(function (total, child) {
-      return total + 1 + child.replyCount;
-    }, 0),
-  });
+  const children = thread.children.map(c => countRepliesAndDepth(c, depth + 1));
+  const replyCount = children.reduce(
+    (total, child) => total + 1 + child.replyCount,
+    0
+  );
+  return {
+    ...thread,
+    ...{
+      children,
+      depth,
+      replyCount,
+    },
+  };
 }
 
-/** Return true if a thread has any visible children. */
+/**
+ * Does this thread have any visible children?
+ *
+ * @param {Thread} thread
+ * @return {boolean}
+ */
 function hasVisibleChildren(thread) {
-  return thread.children.some(function (child) {
+  return thread.children.some(child => {
     return child.visible || hasVisibleChildren(child);
   });
 }
 
 /**
  * @typedef Options
- * @prop {string[]} [selected]
- * @prop {string[]} [forceVisible]
- * @prop {(a: Annotation) => boolean} [filterFn]
- * @prop {(t: Thread) => boolean} [threadFilterFn]
- * @prop {Object} [expanded]
- * @prop {Object} [highlighted]
- * @prop {(a: Annotation, b: Annotation) => boolean} [sortCompareFn]
- * @prop {(a: Annotation, b: Annotation) => boolean} [replySortCompareFn]
+ * @prop {string[]} [selected] - List of currently-selected annotation ids, from
+ *       the data store
+ * @prop {string[]} [forceVisible] - List of ids of annotations that have
+ *       been explicitly expanded by the user, even if they don't
+ *       match current filters
+ * @prop {(a: Annotation) => boolean} [filterFn] - Predicate function that
+ *       returns `true` if annotation should be visible
+ * @prop {(t: Thread) => boolean} [threadFilterFn] - Predicate function that
+ *       returns `true` if the annotation should be included in the thread tree
+ * @prop {Object.<string, boolean>} [expanded] - Map of thread id => expansion state
+ * @prop {Object} [highlighted] - List of ids of annotations that are highlighted
```

Comment:
```suggestion
 * @prop {string[]} [highlighted] - List of ids of annotations that are highlighted
```

Reply:
This reminds me: we should conventionlize on either `{Array<type>}` or `{type[]}` perhaps?